lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -30,20 +31,26 @@
 
 <li class="course-item">
   % if settings.FEATURES.get('ENABLE_VERIFIED_CERTIFICATES'):
-    % if enrollment.mode == "verified":
-      % if verification_status.get('status') in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
-        <% mode_class = " verified" %>
-      % else:
-        <% mode_class = " honor" %>
-      % endif
-    % else:
-      <% mode_class = " " + enrollment.mode %>
-    % endif
+    <% course_verified_certs = CourseMode.get_certificate_display(enrollment.mode, verification_status.get('status')) %>
+    <%
+        display_mode = course_verified_certs.get('display_mode')
+        if display_mode == 'verified':
+             mode_class = 'verified'
+        elif display_mode == 'honor':
+            mode_class = 'honor'
+        elif display_mode == 'audit':
+            mode_class = 'audit'
+        elif display_mode == 'professional':
+            mode_class = 'professional'
+        else:
+            mode_class = ''
```

Comment:
I'm going to echo @wedaly on this one; the `mode_class` is currently the same as the `display_mode`, and I think that use of this large conditional block is an example of optimizing prematurely for the needs of UX.